sdk/reservations/Azure.ResourceManager.Reservations/src/autorest.md

```diff
@@ -118,6 +118,12 @@ directive:
   - from: reservations.json
     where: $.definitions
     transform: >
+      $.ReservationOrderProperties.properties.expiryDate['x-ms-client-name'] = 'ExpireOn';
+      $.ReservationOrderProperties.properties.expiryDateTime['x-ms-client-name'] = 'ReservationExpireOn';
+      $.ReservationsProperties.properties.expiryDate['x-ms-client-name'] = 'ExpireOn';
+      $.ReservationsProperties.properties.expiryDateTime['x-ms-client-name'] = 'ReservationExpireOn';
+      $.ReservationsProperties.properties.purchaseDate['x-ms-client-name'] = 'PurchaseOn';
+      $.ReservationsProperties.properties.purchaseDateTime['x-ms-client-name'] = 'ReservationPurchaseOn';
```

Comment:
Just as we talked offline, these renamings is just going to work around the generation failure, we need to come up with better and more descriptive names for these properties before we merge this PR.